### PR TITLE
librosa-python_speech_features

### PR DESCRIPTION
**Pull request by vsl9**
librosa branch + backward compatibility with python_speech_features configs

### CODE REVIEW

File: open_seq2seq/data/speech2text/speech2text.py [hunk 1]
```diff
@@ -36,15 +37,23 @@ def get_required_params():
...
         'syn_enable': bool,
         'syn_subdirs': list,
         'window_size': float,
         'window_stride': float,
+        'dither': float,
+        'norm_per_feature': bool,
+        'window': ['hanning', 'hamming', 'none'],
+        'num_fft': int,
+        'precompute_mel_basis': bool,
+        'sample_freq': int,
```
Comment (blisc) [inline open_seq2seq/data/speech2text/speech2text.py]: We should add comments explaining what the new parameters do.
Missing 'min_duration', 'dither', 'norm_per_feature', 'window', 'num_fft'

Comment (blisc) [reply open_seq2seq/data/speech2text/speech2text.py]: dither, norm_per_feature, num_fft are explained in get_speech_features_from_file. We should copy and paste here

File: open_seq2seq/data/speech2text/speech_utils.py [hunk 2]
```diff
@@ -130,43 +142,42 @@ def fix_kv(text):
...
+      window_stride (float): stride of analysis window in milli-seconds.
+      augmentation (dict, optional): None or dictionary of augmentation parameters.
+        If not None, has to have 'time_stretch_ratio',
+        'noise_level_min', 'noise_level_max' fields, e.g.::
+          augmentation={
+            'time_stretch_ratio': 0.2,
+            'noise_level_min': -90,
+            'noise_level_max': -46,
+          }
+      window_fn (bool): window function to apply, or None for no window
```
Comment (blisc) [inline open_seq2seq/data/speech2text/speech_utils.py]: should be window (string or None), not window_fn (bool)

File: requirements.txt [hunk 3]
```diff
@@ -1,11 +1,10 @@
 numpy
 nltk==3.2.5
 resampy
-python_speech_features
```
Comment (blisc) [inline requirements.txt]: Should we still keep psf?

File: open_seq2seq/encoders/tdnn_encoder.py [hunk 4]
```diff
@@ -95,6 +98,19 @@ def _encode(self, input_dict):
     """
 
     source_sequence, src_length = input_dict['source_tensors']
+    
+    num_pad = tf.constant(0)
+
+    if isinstance(self._model.get_data_layer(), Speech2TextDataLayer):
```
Comment (blisc) [inline open_seq2seq/encoders/tdnn_encoder.py]: We should do an `if backend is librosa` check here.

File: open_seq2seq/data/speech2text/speech2text.py (hunk 1 above)
Comment (vsl9) [reply open_seq2seq/data/speech2text/speech2text.py]: Fixed

File: open_seq2seq/data/speech2text/speech_utils.py (hunk 2 above)
Comment (vsl9) [reply open_seq2seq/data/speech2text/speech_utils.py]: Fixed

File: requirements.txt (hunk 3 above)
Comment (vsl9) [reply requirements.txt]: let's keep it for some time

File: open_seq2seq/encoders/tdnn_encoder.py (hunk 4 above)
Comment (vsl9) [reply open_seq2seq/encoders/tdnn_encoder.py]: Fixed

**Pull request merged**